### PR TITLE
KAPT: Fix error reporting

### PR DESCRIPTION
**Pull request by gavra0**
When incremental analysis is unable to run, handle
failure gracefully. More info will be provided by
compiler diagnostics.

Fixes KT-36302

### CODE REVIEW

File: plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt [hunk 1]
```diff
@@ -22,24 +22,32 @@ class MentionedTypesTaskListener(
...
-        compilationUnit.typeDecls.forEach {
-            it.accept(treeVisitor, Visibility.ABI)
+            val treeVisitor = TypeTreeVisitor(elementUtils, trees, compilationUnit, structure)
+            compilationUnit.typeDecls.forEach {
+                it.accept(treeVisitor, Visibility.ABI)
+            }
+            cache.addSourceStructure(structure)
+            time += System.currentTimeMillis() - l
+        } catch (t: Throwable) {
+            failureReason = "Running non-incrementally because analyzing ${e.sourceFile.toUri()} failed."
```
Comment (yanex) [inline plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt:49]: Should we print the exception to a debug log so we can understand what went wrong on a user machine?

Comment (gavra0) [reply plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt:49]: We can add that to debug log. I did not want to add it to info/warn as it is mostly obscure compiler-infra exception, but adding it to debug makes sense.

Comment (gavra0) [reply plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt:49]: This is now fixed in https://github.com/JetBrains/kotlin/compare/master...google:prr/gavra/kapt-better-error-message?expand=1 but PR is not updated for some reason.

**Comment by abelkov** [comment]
Fixed in 1.4.0 branch: https://github.com/JetBrains/kotlin/commit/5d1727b68138babb42f85febaaef5a4f5699d9c3

**Pull request closed without merging**